user_management/api/tests/urls.py

```diff
@@ -1,17 +1,20 @@
 from django.conf.urls import include, url
-from django.contrib.auth.views import login
+from django.contrib.auth.views import LoginView
 
 
 urlpatterns = [
     url(r'', include(
-        [
-            url(r'', include('user_management.api.urls')),
-            url(r'', include('user_management.api.urls.verify_email')),
-            url(r'', include('user_management.api.urls.users')),
-            url(r'', include('user_management.api.avatar.urls')),
-            url(r'', include('user_management.ui.urls')),
-        ],
-        namespace='user_management_api'
+        (
+            [
+                url(r'', include('user_management.api.urls')),
+                url(r'', include('user_management.api.urls.verify_email')),
+                url(r'', include('user_management.api.urls.users')),
+                url(r'', include('user_management.api.avatar.urls')),
+                url(r'', include('user_management.ui.urls')),
+            ],
+            'user_management_api',
```

Comment:
This didn't seem like the right thing to do!

Reply:
This does seem odd...

Reply:
I just did what the error from Django suggested!

Reply:
I am going to rework the urls so that each entry point has an `app_name` / `namespace`... resulting in lots of namespaces ... `user_management_api_core`, `user_management_api_verify`, `user_management_api_users`, `user_management_api_avatar` and `user_management_ui`

```
urlpatterns = [
    url(r'', include(
        'user_management.api.urls',
        namespace='user_management_api_core',
    )),
    url(r'', include(
        'user_management.api.urls.verify_email',
        namespace='user_management_api_verify',
    )),
    url(r'', include(
        'user_management.api.urls.users',
        namespace='user_management_api_users',
    )),
    url(r'', include(
        'user_management.api.avatar.urls',
        namespace='user_management_api_avatar',
    )),
    url(r'', include(
        'user_management.ui.urls',
        namespace='user_management_ui',
    )),
    url(r'^login/$', LoginView.as_view(), name='login')
]
```

Any thoughts on the app_names above @meshy @Ian-Foote ?

Reply:
I'm not a fan of django's namespaces. They just seem like more hassle than they're worth. I'd rather just see the URL names have a prefix.

Reply:
We are already using them in the tests and docs and you can no longer use a namespace without an `app_name`.

Prefixing all the urls seems like a bigger change and a bit un-django like.... but  I agree they do seem to be more hassle than they're worth

Reply:
How about #173 ?

Reply:
The alternative is to remove the `user_management_api` namespace altogether and update usages to `reverse` (mostly in tests) the `url`.

 Neither options is very backwardly compatible.